OPSEXP-2528: introduce KEDA scaling auto support for ATS and repo (#1161)
Alfresco/acs-deployment: docs/helm/autoscaling.md

e.g:

```bash
helm install \
  --repo https://kedacore.github.io/charts alfresco-keda keda \
  --namespace keda \
  --version 2.14.2
```

## Alfresco components auto-scaling

### Alfresco Repository

#### Basic (CPU based) scaling for Alfresco repository

Refer to the
[alfresco-repository auto-scaling
documentation](https://github.com/Alfresco/alfresco-helm-charts/blob/main/charts/alfresco-repository/docs/autoscaling.md)
for a detailed guide on Alfresco repository auto-scaling configuration and
implications.

#### KEDA based scaling for Alfrsco repository

To start with, make sure your Kubernetes cluster has KEDA & prometheus installed.
You must also make sure Alfresco repository is setup to expose prometheus
metrics and prometheus has the appropriate scrape configuration.

Refer to the [acs-packaging
doc](https://github.com/Alfresco/acs-packaging/tree/master/docs/micrometer)

The minimum configuration for the Alfresco repository to expose prometheus
metrics should be:

```yaml
alfresco-repository:
  environment:
    CATALINA_OPTS: >-
      -Dmetrics.enabled=true
      -Dmetrics.jvmMetricsReporter.enabled=true
      ...
```

##### Prometheus scaler

> The KEDA based auto scaler relies on the number of Tomcat thread used. By

Reviewer: ```suggestion
The KEDA based auto scaler relies on the number of Tomcat threads used. By
```

default the Alfresco repository image uses up to 200 threads. When the system
consistently uses more than 170 threads, the KEDA scaler will start to scale up
the number of pods. This can be tuned using the
`alfresco-repository.autoscaling.kedaTargetValue` if your image has a
configuration with more or less `maxThreads`.
In the same maner the parameters below can be set:

* `behavior.scaleUp.stabilizationWindowSeconds`: The number of threads used must
  remain above target on average for 30 seconds before a scale up can happen.
* `kedaPollingInterval`: threads are checked every 15 seconds.
* `kedaInitialCoolDownPeriod`: KEDA will wait for 5 minutes before activating
  the scaling object (before no scaling can happen).
* `minReplicas`: The default minimum number of replica count is 1.
* `maxReplicas`: The default maximum number of replica count is 3.

### Alfresco Transform Service

#### Basic (CPU based) scaling for ATS

Refer to the
[alfresco-repository auto-scaling

> documentation](https://github.com/Alfresco/alfresco-helm-charts/blob/main/charts/alfresco-transform-service/docs/autoscaling.md)

Reviewer: link the jekyll version across different repos

* `kedaTargetValue`: new pods will be started when the corresponding message
  queue has more than 10 messages.
* `behavior.scaleUp.stabilizationWindowSeconds`: The number of messages in the
  queue must remain above target on average for 30 seconds before a scale up can
  happen.
* `kedaPollingInterval`: Queues are checked every 15 seconds.
* `kedaInitialCoolDownPeriod`: KEDA will wait for 5 minutes before activating
  the scaling object (before no scaling can happen).
* `kedaCooldownPeriod`: After KEDA has found there is no activity in the
  monitored queue, it will wait for 15 minutes before scaling down the pods to
  0.
* `kedaIdleReplicas`: The default idle replica count is 0 (tears down the
  service).
* `minReplicas`: The default minimum number of replica count is 1.
* `maxReplicas`: The default maximum number of replica count is 3.

> Values mentioned above must be set for each tengine
> `alfresco-transform-service._TENGINE_NAME_.autoscaling` where `_TENGINE_NAME_`
> is one of the following: `imagemagick`, `libreoffice`, `transformmisc`,
> `pdfrenderer` & `tika`.

Scaling replicas down to zero is great when you have workload that is consistent

> enough with long period of inactivity (e.g. overnigh). But it can trigger a

Reviewer: ```suggestion
enough with long period of inactivity (e.g. during the night). But it can trigger a
```

delay for the first requests when the workload starts again (e.g. the morning
after). If you want to avoid scaling down you ATS deployments down to zero and
always have at least one pod up to deal quickly with "sparse" requests just
apply the yaml below for the appropriate scaler object (here for pdf
convertion):

```yaml
alfresco-transform-service:
  pdfrenderer:
    autoscaling:
      kedaIdleReplicas: null
```

**Important**: If you're using a version of the ATS T-router prior to 5.1.3, you
need to set the `kedaIdleReplicas` to `0` for all tengines, otherwise the
T-router will eventually crash.

> If you want to use an external ActiveMQ broker instead of the embeded one

Reviewer: ```suggestion
If you want to use an external ActiveMQ broker instead of the embedded one
```
